### ISSUE TITLE
integralClosure produce normalizations with repeated variable names

### ISSUE THREAD

**Issue by kschwede** [Macaulay2/M2#743]
Consider the following example.

```
C = QQ[B1,B2,B3,B4,B5,B6]
I =  ideal(B4*B5+B1*B6,B1*B4+B2*B4-B3*B6,B1^2+B1*B2+B3*B5,B2*B5^2-B6^2,B1*B2*
       B5+B4*B6,B3*B4^2-B6^2,B3^2*B4-B1*B6-B2*B6,B2*B3*B4-B3^2*B6-B5*B6,B3^3-B1
       *B2-B2^2+B3*B5,B1*B3^2+B1*B5+B2*B5,B1*B2*B3+B3^2*B5+B5^2,B1*B2^2+B2*B3*
       B5+B4^2,B3^2*B5^2+B5^3-B3*B4*B6,B2^3*B4-B2^2*B3*B6-B3^2*B5*B6-B4^3-B5^2*
       B6)
D = C/I

integralClosure D
```
outputs 
```
QQ[w_(0,0), w_(0,0), B1, B2, B3, B4, B5,
       B6]/(B4*B5+B1*B6,B1*B4+B2*B4-B3*B6,B1^2+B1*B2+B3*B5,w_(0,0)*B6-B3*B4,w_(
       0,0)*B5+B1*B3,w_(0,0)*B4-B6,w_(0,0)*B3-B1-B2,w_(0,0)*B2-B3^2-B5,w_(0,0)*
       B1+B5,w_(0,0)^2-B3,w_(0,0)*B6-B2*B5,w_(0,0)*B5-B6,w_(0,0)*B4+B1*B2,w_(0,
       0)*B1+B4,w_(0,0)^2-B2)
```

Note that the variable w_(0,0) appears twice...

However, if I rename the new variables, only the first new variable is renamed.

```
i5 : toString integralClosure(D, Variable=>symbol TTT)

o5 = QQ[TTT_(0,0), w_(0,0), B1, B2, B3, B4, B5,
     B6]/(B4*B5+B1*B6,B1*B4+B2*B4-B3*B6,B1^2+B1*B2+B3*B5,w_(0,0)*B6-B3*B4,w_(0,0
     )*B5+B1*B3,w_(0,0)*B4-B6,w_(0,0)*B3-B1-B2,w_(0,0)*B2-B3^2-B5,w_(0,0)*B1+B5,
     w_(0,0)^2-B3,TTT_(0,0)*B6-B2*B5,TTT_(0,0)*B5-B6,TTT_(0,0)*B4+B1*B2,TTT_(0,0
     )*B1+B4,TTT_(0,0)^2-B2)
```

Finally, note that trying to run this example with the alternate strategy outputs an error.

`
integralClosure(D, Strategy=>{RadicalCodim1})
stdio:9:1:(3): error: map: multiple variables would map to the same variable, by name
`

**Comment by pzinn** [comment]
this is not very hard to fix: one needs to make makeS2 keep track of the optional Variable, and then start nsteps at 1, not 0. 
but then someone has to go through all the examples and fix the expected answers...
I know Mike and David are busy men; I'll do it if nobody else will!

**Comment by DanGrayson** [comment]
What do you think the problem is, then?

**Comment by pzinn** [comment]
The problem is that integralClosure first calls makeS2, which adds some
variables, and then does it own addition of variables to normalize. both
functions have the same default variable name (and at the moment, makeS2
doesn't allow to change the default), hence the conflict of names. since
makeS2 is only one step, the first index is always 0, so a simple solution
is to make the step number of integralClosure start at 1.
actually, I have already implemented it, so unless there's any objection
I'll make a pull request in a day or two after proper testing/documenting.

**Comment by eisenbud** [comment]
Note that there's a different "S2" routine in CompletIntersectionResolutions.m2 that takes a module and returns a module. It might be good to combine them -- perhaps even putting them in the core (but not for this release!)
--
David Eisenbud
Director, Mathematical Sciences Research Institute; and
Professor of Mathematics,University of California, Berkeley
www.msri.org/~de

**Comment by DanGrayson** [comment]
Okay, sounds good, Paul.

**Comment by DanGrayson** [comment]
Fixed in #751
